### ISSUE TITLE
qt_gui.runAskYesNoCancelDialog Prints Error Messages

### ISSUE THREAD

**Issue by tbpassin** [leo-editor/leo-editor#2943]
When I get a AskYesNoCancelDialog (usually, to ask if I want to close without saving), I've noticed that I get an error message:

```
Could not parse stylesheet of object QMessageBox(0x278a47d7120)
```

I get similar messages for the file dialog.

At first I thought there was something wrong with the stylesheets I use, but instead, it's the stylesheet that is fed to the dialog:

```
    stylesheet = getattr(c, 'active_stylesheet', None)
    if stylesheet:
        dialog.setStyleSheet(stylesheet)
```

The dialogs work as intended, but it's annoying to keep getting the error messages.

This stylesheet is the raw theme stylesheet that contains _@parameters_.  I'm fairly sure that the dialogs need a stylesheet that has the parameters replaced with the actual values.  Here's a fragment that the dialog actually receives:

```
* {
font-family: @font-family-ui;
font-size: @font-size-ui;
color: @text-foreground;
background: @dark-base03;
}
```

Other dialogs probably have this problem too, but I haven't checked them.

**Comment by tbpassin** [comment]
The following code fixes the issue for the _runAskYesNoCancel dialog_:

```
    ssm = g.app.gui.styleSheetManagerClass(c)
    w = ssm.get_master_widget()
    sheet = w.styleSheet()
    if sheet:
        dialog.setStyleSheet(sheet)
```

[existing code that causes error messages:
```
    stylesheet = getattr(c, 'active_stylesheet', None)
    if stylesheet:
        dialog.setStyleSheet(stylesheet)
```
]

I haven't looked through the other dialogs to see if they use the same code; I'd guess they do.

Here's how I've been testing - it's the easiest, quickest way I've come up with:

1. From an open Leo session, select the menu item _Settings/Open A Theme File_.
2. Pick any theme you like; a new session opens with the chosen theme.
3. Make any simple change to the outline.  I have made F9 to be a hot key to set a mark, so I just press that.
4. Close the Leo window (or just the theme's tab).  Leo will put up the dialog asking if you want to save the changed outline.  At this time the error message will get printed to the console (or not, if the code were fixed).

**Comment by edreamleo** [comment]
@tbpassin Good catch. This bug has likely been around for a long time. Please file a PR. The fix should be in Leo 6.7.1.

**Comment by tbpassin** [comment]
It looks like it was never necessary to set the stylesheet for the dialogs at all:

- The QMessageBox always uses the theme's colors even if the message box's stylesheet is not set during the Open call (tested with new, default, and very old themes like Swot and PaperLight);

- The QFileDialog never uses the theme colors even when its stylesheet is set correctly.

I'll check about the other dialogs.

**Comment by edreamleo** [comment]
@tbpassin I would have thought Leo processes the stylesheets when loading a .leo file. I'll check later today to see if my memory is correct.  In any case, handling individual dialogs separately seems like a dubious plan.

**Comment by tbpassin** [comment]
I forgot to say that _runAskLeoIDDialog_ can't be set to use the theme's CSS because it is not passed c when invoked.  But at least it does not print an error message about not being able to parse the stylesheet.

I think this could probably be fixed by including c as one of the invocation args.  It would need to be defaulted so that this function's call signature would not be changed.  The PR does not address this.

Maybe we don't care very much if the askLeoID dialog uses the theme's css or not ... although I think the default appearance looks a little clunky.

**Comment by edreamleo** [comment]
Closed via PR #2946.
